### PR TITLE
feat(schemantic): add serialize to convert typed values to JSON

### PR DESCRIPTION
**Pull request by pavelgj**
Add `SchemanticType.serialize` as the inverse of `parse`, converting a typed value back into plain JSON. The default implementation handles scalars, lists, maps, and any object exposing a `toJson()` method recursively. Add an optional `serialize` callback to `SchemanticType.from` for custom encodings when bridging your own types.

### CODE REVIEW

File: packages/schemantic_builder/test/schema_generator_test.dart [hunk 1]
```diff
@@ -288,6 +288,13 @@ Future<void> _testBuilderWithNoFail(
     outputs: outputs,
     onLog: (log) {
       if (log.level >= Level.WARNING) {
+        // Ignore the environmental warning emitted when the running SDK's
```
Comment (pavelgj) [inline packages/schemantic_builder/test/schema_generator_test.dart:291]: this is to fix a CI failure on dart dev version....

File: packages/schemantic/lib/schemantic.dart [hunk 2]
```diff
@@ -149,16 +149,51 @@ abstract class SchemanticType<T> {
...
+  /// `String`, `num`, `bool`, or `null`) - the inverse of [parse].
+  ///
+  /// The default implementation handles scalars, lists, and maps recursively,
+  /// and delegates to `toJson()` for any other object (all generated schemantic
+  /// types expose one). Override this for custom encodings.
+  Object? serialize(T value) => _serializeToJson(value);
+
+  /// Recursively converts a JSON-shaped [value] (scalar, `List`, `Map`, or an
+  /// object with a `toJson()` method) into plain JSON.
+  static Object? _serializeToJson(Object? value) {
```
Comment (cabljac) [inline packages/schemantic/lib/schemantic.dart:177]: Looks like the result of `toJson()` is returned as-is, without recursing into it. A hand-rolled `toJson()` returning e.g. `{'child': someDomainObject}` would leak non-JSON values despite the doc promising plain JSON. Intentional trade-off, or should this be `_serializeToJson((value as dynamic).toJson())`?

**Review by cabljac — APPROVED** [review]
One comment, no other blockers

File: packages/schemantic/lib/schemantic.dart (hunk 2 above)
Comment (pavelgj) [reply packages/schemantic/lib/schemantic.dart:177]: Good catch, wasn't intentional. Fixed it to recurse into the toJson() result so nested domain objects get normalized to plain JSON. Kept the NoSuchMethodError catch scoped to just the toJson() lookup so an error thrown inside a valid nested toJson() doesn't get mislabeled. Added tests for both the nested-normalization and nested-non-serializable cases. (One edge case: a self-referential toJson() would now stack-overflow instead of silently leaking, but that's already a broken toJson().)
